fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java

```diff
@@ -584,7 +587,11 @@ public void checkTimeout(long now) {
                 killConnection = true;
             }
         } else {
-            if (delta > sessionVariable.getQueryTimeoutS() * 1000L) {
+            long timeoutSecond = sessionVariable.getQueryTimeoutS();
+            if (isPending) {
```

Comment:
isPending means the query is in the queue, wait to be scheduled ? is GlobalVariable.getQueryQueuePendingTimeoutSecond() return a constant or a increasing pending time? I think the latter would be a  better choice.

Reply:
1. Yes, `isPending` means the query is in the queue and wait to be scheduled.
2. `GlobalVariable.getQueryQueuePendingTimeoutSecond() ` returns a constant threshold. The query will be failed, if it has been in the queue for more than `timeout`. Here is to avoid making query failed earlier if `query_timeout < pending_timeout`.